ir/state.cpp

```diff
@@ -143,10 +238,24 @@ static expr not_poison_except_padding(const Type &ty, const expr &np) {
   return result;
 }
 
-const StateValue& State::getAndAddPoisonUB(const Value &val) {
-  auto &v = (*this)[val];
-  if (!analysis.non_poison_vals.insert(&val).second)
-    return v;
+const StateValue&
+State::getAndAddPoisonUB(const Value &val, bool undef_ub_too) {
+  auto &sv = (*this)[val];
+
+  if (!analysis.non_poison_vals.insert(&val).second && !undef_ub_too)
+    return sv;
+
+  expr v = sv.value;
+
+  if (undef_ub_too) {
+    auto inserted = analysis.non_undef_vals.find(&val);
```

Comment:
inserted is not a good choice of var name here. can it be simply `I` for iterator?

Reply:
also, you can use `try_emplace(&val)`, for example, to avoid the second map lookup below.

Reply:
Updated the var name to I.
Regarding try_emplace: `strip_undef_and_add_ub` call below calls State::operator[] as well, and it looks through `analysis.non_undef_vals`. try_emplace causes an interaction between two.
To avoid unnecessary changes at State::operator[], simply find was used instead.

Reply:
sounds good.